src/Stan/Analysis/Analyser.hs

```diff
@@ -30,19 +30,13 @@ import qualified Data.Set as Set
 import qualified Slist as S
 
 
-{- | Contains analyser function to run on 'HieFile's.
+{- | Create analysing function for 'Inspection' by pattern-matching
+over 'InspectionAnalysis'.
 -}
-data Analyser = Analyser
-    { analyserInspectionId :: !(Id Inspection)
-    , analyserFunction     :: !(HieFile -> Observations)
-    }
-
--- | Smart constructor for 'Analyser' creation of partial functions.
-mkNameMetaAnalyser :: Id Inspection -> NameMeta -> Analyser
-mkNameMetaAnalyser insId nameMeta = Analyser
-    { analyserInspectionId = insId
-    , analyserFunction = analyseNameMeta insId nameMeta
-    }
+analysisByInspection :: Inspection -> HieFile -> Observations
+analysisByInspection Inspection{..} = case inspectionAnalysis of
+    FindName nameMeta -> analyseNameMeta inspectionId nameMeta
+    Infix             -> const mempty  -- TODO: not yet implemented
```

Comment:
I've created two constructors, so HLint won't suggest to use `newtype` instead of `data`